### DISCUSSION TITLE
initial page loader

### DISCUSSION THREAD

**Discussion by yogeshicube** [nuxt/nuxt#22243] [General]
hey, :)

I am creating a NUXT js  app and would like to add in an initial homepage loader i.e. when the user first arrives on the app a preloader page should show until the homepage content has fully rendered.

Any suggestions / experience with this ?

**Comment by picunada** [comment]
Hi, @yogeshicube.

You can use onMounted hook.
If you have specific content/script loading on client side or maybe you want your page to have some preloader as a design part of your app, you can create variable with useState composable, depending on state you can show/hide preloader.

If you have some extensive code to load on client side as some WebGL stuff, and it takes some time to appear on the page, you can set state to true when your WebGL code have executed.

Example with WebGL class:
```vue
<!-- Preloader component -->
<script setup lang="ts">

const preloaded = useState('loaded')

// You can watch for state
watch(preloaded, (v) => {
  if (v) {
    // You can run some code if the state changes to true
    // for example gsap timeline start playing
  }
})
</script>
```

```js
// WebGL class with extensive code
export default class Sketch {
  public loaded: Ref<boolean> = useState('loaded', () => false)

  constructor() {
    ...
    // Set state after code was executed
    this.loaded.value = true
  }
}
```

**Comment by unshame** [comment]
If you're using ssr: false and want a loader to show up before nuxt is initialized, I think this is the option for you: https://nuxt.com/docs/api/configuration/nuxt-config#spaloadingtemplate
